### DISCUSSION TITLE
How to apply custom CSS to WYSISYG Editor?

### DISCUSSION THREAD

**Discussion by TheOneAndOnlyGarcia** [directus/directus#4117] [Archive]
Hey, 

can someone help me out how to apply custom styles to the **rendered preview** of the WYSIWYG Editor? 

Inside WYSIWYG Editor (source code view):

```html
<p>
This is <span class="important">very important</span>!
</p>
```

What CSS Classes do i need to overwrite or subclass to insert this into the directus app code? Like copy paste into App Overrides -> Custom CSS?
```css
.important {
  background-color: yellow;
}
```

Thanks a lot!

**Comment by rijkvanzanten** [comment]
The CSS in the wysiwyg is rendered in an iFrame by TinyMCE, so it's not as easy as you'd hope to override that. That being said, the wysiwyg interface has a tinymce-overrides option you can use to manually override [the `content_style` option](https://github.com/directus/directus/blob/main/app/src/interfaces/wysiwyg/wysiwyg.vue#L143) of TinyMCE.

**Comment by loomchild** [comment]
I was able to override the style via Options Override `content_style` setting ([TinyMCE docs](https://www.tiny.cloud/docs/configure/content-appearance/)):

```
{
    "content_style": "blockquote.icon-information { color: red; }"
}
```

However, It replaces the whole styling of the WYSIWYG element. Is it the expected behavior?

If yes, then can you please suggest a way to preserve the original styling, just extending it? I tried using `content_css` option and that seems to maintain the original style and fetch my custom file, but I don't know where should I put the file in Directus to make it accessible via a constant URL. Is it possible via an extension, perhaps?

BTW, the above link doesn't work anymore. Perhaps [this is the new one](https://github.com/directus/directus/tree/main/app/src/interfaces/input-rich-text-html)?

**Reply by azrikahar** [reply]
> However, It replaces the whole styling of the WYSIWYG element. Is it the expected behavior?

Unfortunately it is, since this would indeed override the original styling here:

https://github.com/directus/directus/blob/ce54bafa4b64f801f8fce0f13da7ca51a75ef0ac/app/src/interfaces/input-rich-text-html/input-rich-text-html.vue#L331

which comes from [the styles defined in this function here](https://github.com/directus/directus/blob/main/app/src/interfaces/input-rich-text-html/get-editor-styles.ts).

> If yes, then can you please suggest a way to preserve the original styling, just extending it?

One alternative would be to re-use the styles from the above function, but this would mean you'll need to manually replace those cssVar definitions and prolly lose the font props flexibility, so understandably not ideal.

Perhaps as you mentioned, using `content_css` may be the "easier" option. You can create a [custom endpoint extension](https://docs.directus.io/extensions/endpoints/) to make the custom css accessible via a URL of your choice. 

> BTW, the above link doesn't work anymore. Perhaps [this is the new one](https://github.com/directus/directus/tree/main/app/src/interfaces/input-rich-text-html)?

Yes that is correct 👍 likely moved due to the port to Vue 3 a while ago.

---

Off-topic: contributed to your volume-backup repo a while back, didn't expect to bump into you here. Small world!

**Reply by loomchild** [reply]
Thanks very much for your rapid reply. Small world indeed!

**Comment by Warain811** [comment]
Hello, does using content_css and the custom endpoint extension still work? If so, how should the custom endpoint extension look like? Thank you.

**Reply by bootleq** [reply]
It is still feasible, I have created an example project:
https://github.com/bootleq/directus-extension-tinymce-endpoint-example

**Reply by Warain811** [reply]
Thanks! 🙂
